Projects/Core/Common/Utilities/DebugLogger.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  DebugLogger.swift
+//  Common
+//
+//  Created by GREEN on 6/14/24.
+//  Copyright © 2024 com.mashup.gabbangzip. All rights reserved.
+//
+
+import Foundation
+
+public let logger = DebugLogger()
+
+public class DebugLogger {
+  private static let fileLogQueue = DispatchQueue(label: "com.mashup.gabbangzip.file_log_queue")
+  
+  public init() { }
+  
+  public func debug(
+    _ message: String = "",
+    filePath: String = #file,
+    funcName: String = #function,
+    line: Int = #line
+  ) {
+    printLog(
+      message,
+      symbol: "🍀",
+      filePath: filePath,
+      funcName: funcName,
+      line: line
+    )
+  }
+  
+  public func warn(
+    _ message: String = "",
+    filePath: String = #file,
+    funcName: String = #function,
+    line: Int = #line
+  ) {
+    printLog(
+      message,
+      symbol: "🚧",
+      filePath: filePath,
+      funcName: funcName,
+      line: line
+    )
+  }
+  
+  public func error(
+    _ message: String = "",
+    filePath: String = #file,
+    funcName: String = #function,
+    line: Int = #line
+  ) {
+    printLog(
+      message,
+      symbol: "❌",
+      filePath: filePath,
+      funcName: funcName,
+      line: line
+    )
+  }
+  
+  private func printLog(
+    _ message: String,
+    symbol: String,
+    filePath: String,
+    funcName: String,
+    line: Int
+  ) {
+    #if DEBUG
+    let fileName = DebugLogger.sourceFileName(filePath: filePath)
+    let formatter = DateFormatter.iso8601
```

Comment:
P3:
```suggestion
    let formatter = ISO8601DateFormatter()
```
제 Xcode에선 `Type 'DateFormatter' has no member 'iso8601'`라는 컴파일 에러가 발생하네요. 새로운 타입(`ISO8601DateFormatter`)을 제안드립니다🙋‍♂️

Reply:
앗ㅋㅋ 제가 DataFormatter 코드를 만들어넣고 테스트하다 안올렸네요.
DataFormatter를 확장한 코드를 이용할께요.
이유는 TimeZone 및 Locale 등 보다 더 한번에 관리하기 편하여 앞으로도 DateFormat할 경우 해당 extension파일에서 작업해도 될거라 생각합니다.